smartsim/_core/generation/generator.py

```diff
@@ -240,119 +269,60 @@ def _execute_commands(cls, cmd_list: CommandList) -> None:
         """
-        if files is None:
-            return None
         cmd_list = CommandList()
-        for src in files.copy:
-            cmd = Command(
-                [
-                    sys.executable,
-                    "-m",
-                    "smartsim._core.entrypoints.file_operations",
-                    "copy",
-                    src,
-                ]
-            )
-            destination = str(dest)
-            if os.path.isdir(src):
-                base_source_name = os.path.basename(src)
-                destination = os.path.join(dest, base_source_name)
-                cmd.append(str(destination))
-                cmd.append("--dirs_exist_ok")
-            else:
-                cmd.append(str(dest))
-            cmd_list.commands.append(cmd)
+        for file in files:
+            cmd_list.append(file.format(context))
         return cmd_list
 
     @staticmethod
     def _symlink_files(
-        files: t.Union[EntityFiles, None], dest: pathlib.Path
-    ) -> t.Optional[CommandList]:
-        """Build command to symlink files/directories from specified paths to a destination directory.
-
-        This method creates commands to symlink files/directories from the source paths provided in the
-        `files` parameter to the specified destination directory. If the source is a directory,
-        it copies the directory while allowing existing directories to remain intact.
+        files: t.List[SymlinkOperation], context: GenerationContext
+    ) -> CommandList:
+        """Build commands to symlink files/directories from specified source paths
+        to an optional destination in the run directory.
 
-        :param files: An EntityFiles object containing the paths to symlink, or None.
-        :param dest: The destination path to the Job's run directory.
-        :return: A CommandList containing the symlink commands, or None if no files are provided.
+        :param files: A list of SymlinkOperation objects
+        :param context: A GenerationContext object that holds the Job's run directory
+        :return: A CommandList containing the symlink commands
         """
-        if files is None:
-            return None
         cmd_list = CommandList()
-        for src in files.link:
-            # Normalize the path to remove trailing slashes
-            normalized_path = os.path.normpath(src)
-            # Get the parent directory (last folder)
-            parent_dir = os.path.basename(normalized_path)
-            new_dest = os.path.join(str(dest), parent_dir)
-            cmd = Command(
-                [
-                    sys.executable,
-                    "-m",
-                    "smartsim._core.entrypoints.file_operations",
-                    "symlink",
-                    src,
-                    new_dest,
-                ]
-            )
-            cmd_list.append(cmd)
+        for file in files:
+            cmd_list.append(file.format(context))
         return cmd_list
```

Comment:
Ditto "style nit to be a bit more declarative" comment here
```suggestion
        return CommandList([file.format(context) for file in files])
```